spark/src/test/java/org/apache/iceberg/spark/data/TestSparkAvroUnions.java

```diff
@@ -121,25 +125,76 @@ public void writeAndValidateOptionalComplexUnion() throws IOException {
         .project(expectedSchema)
         .build()) {
       rows = Lists.newArrayList(reader);
+
+      Assert.assertEquals("foo", rows.get(0).getStruct(0, 2).getString(1));
+      Assert.assertEquals(1, rows.get(1).getStruct(0, 2).getInt(0));
```

Comment:
can we verify it is `[foo, null]` and `[1, null]`  (here and everywhere; might worth a verification function)?

Reply:
and may be also verify the size is 2?

Reply:
Added